src/_gettsim/transfers/rente.py

```diff
@@ -19,8 +19,181 @@ def sum_ges_rente_priv_rente_m(priv_rente_m: float, ges_rente_m: float) -> float
+    bruttorente_basisbetrag_m
+        See :func:`bruttorente_basisbetrag_m`.
+    bruttolohn_m
+        See basic input variable :ref:`bruttolohn_m <bruttolohn_m>`.
+    höchster_bruttolohn_letzte_15_jahre_m
+        See :func:`höchster_bruttolohn_letzte_15_jahre_m`.
+    ges_rente_params
+        See params documentation :ref:`ges_rente_params <ges_rente_params>`.
+
+    Returns
+    -------
+
+    """
+    # No deduction because of age or low earnings
+    if (alter >= ges_rente_regelaltersgrenze) or (
+        bruttolohn_m <= ges_rente_params["hinzuverdienstgrenze"] / 12
+    ):
+        zahlbetrag_nach_einkommensanrechung = bruttorente_basisbetrag_m
+    # Basis deduction of 40%
+    else:
+        zahlbetrag_nach_einkommensanrechung = max(
+            bruttorente_basisbetrag_m
+            - ges_rente_params["abzugsrate_hinzuverdienst"]
+            * (bruttolohn_m - ges_rente_params["hinzuverdienstgrenze"] / 12),
+            0.0,
+        )
+
+    # Additional 100% deduction if earnings plus pension exceed Hinzuverdienstdeckel
+    einkommen_über_hinzuverdienstdeckel = (
+        zahlbetrag_nach_einkommensanrechung
+        + bruttolohn_m
+        - höchster_bruttolohn_letzte_15_jahre_m
+    )
+    if (
+        einkommen_über_hinzuverdienstdeckel > 0
+        and alter <= ges_rente_regelaltersgrenze
+        and bruttolohn_m > 0
+    ):
+        out = max(bruttorente_basisbetrag_m - einkommen_über_hinzuverdienstdeckel, 0.0)
+    else:
+        out = zahlbetrag_nach_einkommensanrechung
+
+    return out
+
+
+@policy_info(
+    end_date="2016-12-31",
+    name_in_dag="bruttorente_m",
+    params_key_for_rounding="ges_rente",
+)
+def bruttorente_mit_harter_hinzuverdienstgrenze_m(
+    alter: int,
+    ges_rente_regelaltersgrenze: float,
+    bruttolohn_m: float,
+    bruttorente_basisbetrag_m: float,
+    ges_rente_params: dict,
+) -> float:
+    """Pension benefits after earnings test for early retirees.
+
+    If earnings are above an earnings limit, the pension is fully deducted.
+
+    Parameters
+    ----------
+    alter
+        See basic input variable :ref:`alter <alter>`.
+    ges_rente_regelaltersgrenze
+        See :func:`ges_rente_regelaltersgrenze`.
+    bruttolohn_m
+        See basic input variable :ref:`bruttolohn_m <bruttolohn_m>`.
+    bruttorente_basisbetrag_m
+        See :func:`bruttorente_basisbetrag_m`.
+    ges_rente_params
+        See params documentation :ref:`ges_rente_params <ges_rente_params>`.
+
+    Returns
+    -------
+
+    """
+    # TODO (@MImmesberger): Use age with monthly precision.
+    # https://github.com/iza-institute-of-labor-economics/gettsim/issues/781
+    if (alter >= ges_rente_regelaltersgrenze) or (
+        bruttolohn_m <= ges_rente_params["hinzuverdienstgrenze"] / 12
+    ):
+        out = bruttorente_basisbetrag_m
+    else:
+        out = 0.0
+
+    return out
+
+
+def bruttorente_basisbetrag_m(
     ges_rente_zugangsfaktor: float,
     entgeltp_ost_update: float,
```

Comment:
Apologies I this comes up only now: Why is this the "_update" version? AFAIU, the update is what happens because of *concurrent* wage earnings. How can they affect pension payments in real time?

Would affect all tests added here and probably more.

Reply:
Oh yes this bug was in there for ages. Fixed!

Reply:
Cool, thanks! While you are at it, could you rename in `entgeltp_..._updated` ? Otherwise, not so obvious whether we are talking about the sum or the diff.

Reply:
Yes, makes sense